### DISCUSSION TITLE
Offer a checkbox for generate `.git-blame-ignore-revs` in rebase (or squash) merge

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#16880] [Pull Requests]
GitHub has a feature [Rebase and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#rebase-and-merge-your-pull-request-commits) and one for [Squash and merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits).

GitHub has a feature [Ignore commits in the blame view](https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view) which honors SHAs listed in `.git-blame-ignore-revs`.

I regularly create PRs to repositories that consist solely of spelling fixes. These commits would be great candidates for the `.git-blame-ignore-revs` file. But any repository that uses `Rebase and merge` (or `Squash and merge`) makes it such that I can't actually offer a thing for `.git-blame-ignore-revs` because the commit SHA(s) to be added to the repository won't be known until after the rebase (or squash) and merge happens.

Today, to address this, I basically have to make two PRs, one for the spelling fixes. Then after they're rebased (or squashed) and merged, I have to go back and create a new PR to ignore the SHA(s) created by that first PR. There's less incentive for me to do this step, and it looks like I'm gaming the system by creating additional commits/PRs (I really don't care, but people regularly accuse me of doing things like that).

The easiest (and safest) way for GitHub to help would be to offer a:
 - [x] Create entries in `.git-blame-ignore-revs` for commit(s) in this PR

**Comment by spenserblack** [comment]
On the other hand, if that checkbox existed, I think beginners who don't really understand what `.git-blame-ignore-revs` is will click on it, and accidentally ignore commits that they should *not* ignore. Also, a PR might include *some* commits that should be ignored, and some that should not. E.g. a PR that formats a file *then* updates its behavior.

Instead of a checkbox, I think that a PR that adds or modifies a `.git-blame-ignore-revs` could warn the user when they select the "squash" or "rebase" merge strategies. Just something like "this merge strategy may result in invalid entries in `.git-blame-ignore-revs`", prompting the merging user to think a bit more about what they want to do (the first ask in #136477).
